test/common/dns.js

```diff
@@ -0,0 +1,290 @@
+'use strict';
+
+// Naïve DNS parser/serializer.
+
+require('../common');
+const assert = require('assert');
+const os = require('os');
+
+const types = {
+  A: 1,
+  AAAA: 28,
+  NS: 2,
+  CNAME: 5,
+  SOA: 6,
+  PTR: 12,
+  MX: 15,
+  TXT: 16,
+  ANY: 255
+};
+
+const classes = {
+  IN: 1
+};
+
+function readDomainFromPacket(buffer, offset) {
+  assert.ok(offset < buffer.length);
+  const length = buffer[offset];
+  if (length === 0) {
+    return { nread: 1, domain: '' };
+  } else if ((length & 0xC0) === 0) {
+    offset += 1;
+    const chunk = buffer.toString('ascii', offset, offset + length);
+    // Read the rest of the domain.
+    const { nread, domain } = readDomainFromPacket(buffer, offset + length);
+    return {
+      nread: 1 + length + nread,
+      domain: domain ? `${chunk}.${domain}` : chunk
+    };
+  } else {
+    // Pointer to another part of the packet.
+    assert.strictEqual(length & 0xC0, 0xC0);
+    // eslint-disable-next-line
```

Comment:
Eslint is unhappy about both the missing space before `~` and after `&`, so it’s two separate rules.

> If `&~ 0xC000` on the next line is changed to `& ~0xC000`, then we don't need this ESLint disable comment.

Right, but I’d consider `&~` more idiomatic for bit-twiddling code (because it removes an extra thought iteration, from “bitwise and with some oddly generated mask” to “it’s just removing those bits”).